File: demo/HelloSwift/HelloSwift/Sort/SelectionSort.swift
//
//  SelectionSort.swift
//  HelloSwift
//
//  Created by muhlenXi on 2019/9/19.
//  Copyright © 2019 muhlenXi. All rights reserved.
//

import Foundation

func selectionSort(array: inout [Int]){
    guard array.count > 1 else {
        return
    }
    
    for i in 0..<array.count {
        var minValueIndex = i
        for j in i+1..<array.count {
            if array[j] < array[minValueIndex] {
                minValueIndex = j
            }
        }
        if minValueIndex != i {
            let temp = array[i]
            array[i] = array[minValueIndex]
            array[minValueIndex] = temp
        }
    }
}

File: demo/HelloJava/src/Sort/SelectionSort.java
package Sort;

public class SelectionSort {
    public static void main() {
        int[] array = {3, 5, 2, 4 , 1};

        int[] sorted = selectionSort(array);
        for(int value: sorted) {
            System.out.println(value);
        }

    }

    public static int[] selectionSort(int[] array) {
        if(array.length < 2) {
            return array;
        }

        int[] result = array;
        for(int i = 0; i < result.length; i ++) {
            int minValueIndex = i;
            for(int j = i + 1; j < result.length; j++) {
                if (result[j] < result[minValueIndex]) {
                    minValueIndex = j;
                }
            }
            if(minValueIndex != i) {
                int temp = result[minValueIndex];
                result[minValueIndex] = result[i];
                result[i] = temp;
            }
        }
        return result;
    }
}
